scripts/generate_records/generate_db_records.js

```diff
@@ -0,0 +1,403 @@
+ *
+ * @param {BatchParams} params
+ * @returns {Promise<{
+ *   granuleCumulusIds: Array<number>
+ *   fileCumulusIds: Array<number>
+ *   executionCumulusIds: Array<number>
+ * }>}
+ */
+const uploadDataBatch = async ({
+  knex,
+  collectionCumulusId,
+  providerCumulusId,
+  filesPerGranule,
+  granulesPerBatch,
+  executionsPerBatch,
+  models,
+  swallowErrors,
+}) => {
+  const granuleCumulusIds = await loadGranules(
+    knex,
+    collectionCumulusId,
+    providerCumulusId,
+    granulesPerBatch,
+    models.granuleModel,
+    swallowErrors
+  );
+  const fileCumulusIds = [];
+  for (const granuleCumulusId of granuleCumulusIds) {
+    fileCumulusIds.push(
+      await loadFiles(knex, granuleCumulusId, filesPerGranule, models.fileModel, swallowErrors)
+    );
+  }
+  const executionCumulusIds = await loadExecutions(
+    knex,
+    collectionCumulusId,
+    executionsPerBatch,
+    models.executionModel,
+    swallowErrors
+  );
+  await loadGranulesExecutions(
+    knex,
+    granuleCumulusIds,
+    executionCumulusIds,
+    models.geModel,
+    swallowErrors
+  );
+  return {
+    granuleCumulusIds,
+    fileCumulusIds: fileCumulusIds.flat(),
+    executionCumulusIds,
+  };
+};
+
+/**
+ * create a generator Object that pretends to be an Iterable
+ * this is to allow pmap to use this data without holding the entire (potentially very large)
+ * set of batch params for more than the currently running threads
+ *
+ * @param {object} params
+ * @param {Knex} params.knex
+ * @param {number} params.numberOfGranules
+ * @param {number} params.collectionCumulusId
+ * @param {number} params.providerCumulusId
+ * @param {number} params.filesPerGranule
+ * @param {number} params.granulesPerBatch
+ * @param {number} params.executionsPerBatch
+ * @param {ModelSet} params.models
+ * @param {boolean} params.variance
+ * @param {boolean} params.swallowErrors
+ * @returns {Iterable<BatchParams>}
+ */
+
+const getBatchParamGenerator = ({
+  knex,
+  numberOfGranules,
+  collectionCumulusId,
+  providerCumulusId,
+  filesPerGranule,
+  granulesPerBatch,
+  executionsPerBatch,
+  models,
+  variance,
+  swallowErrors = false,
+}) => {
+  if (granulesPerBatch < 1) {
+    throw new Error('granulesPerBatch must be set to >=1');
+  }
+  /**
+   * @yields {BatchParams}
+   */
+  function* batchParamGenerator() {
+    const bar = new cliProgress.SingleBar(
+      { etaBuffer: numberOfGranules / 10 }
+    );
+    bar.start(numberOfGranules, 0);
+
+    const standardParams = {
+      knex,
+      collectionCumulusId,
+      providerCumulusId,
+      filesPerGranule,
+      granulesPerBatch,
+      executionsPerBatch,
+      models,
+      swallowErrors,
+    };
+
+    if (variance) {
+      //asking for variance adds some noise to batch executions vs granules
+      let _granulesPerBatch = 1;
+      let _executionsPerBatch;
+      for (let i = 0; i < numberOfGranules; i += _granulesPerBatch) {
+        _granulesPerBatch = granulesPerBatch + randomInt(6);
```

Comment:
I thought the utilization of the randomInt, was dependent on if variance is true. if that's the case, there should be some sort of if statement here, and I would think the _granulesPerBatch and _executionsPerBatch above should have the default value set to the non underscored value.